frame/staking/src/tests.rs

```diff
@@ -4667,6 +4667,49 @@ fn migrate_era_should_handle_errors_2() {
 	});
 }
 
+#[test]
+fn offences_weight_calculated_correctly() {
```

Comment:
I don't mind this but I think this logic should be mocked in offences as well, namely the fact that it return some weight and ideally a test to show if we overflow on weight, we keep some of the offences and then we re-try in the next block.

Reply:
I added a new test which checks the `WeightSoftLimit`